Makefile

```diff
@@ -3,7 +3,12 @@
 # To re-generate a bundle for another specific version without changing the standard setup, you can:
 # - use the VERSION as arg of the bundle target (e.g make bundle VERSION=0.0.2)
 # - use environment variables to overwrite this value (e.g export VERSION=0.0.2)
-VERSION ?= 0.0.1
+VERSION ?= 0.0.3
+IMG ?= quay.io/vhire/opendatahub-operator:rearch-$(VERSION)
+IMAGE_BUILDER ?= docker
```

Comment:
Can we default to podman?

Reply:
I think we should avoid that. `podman-docker` can be used with podman as a drop-in replacement, while there is nothing in the other direction.

Reply:
Could you elaborate on `podman-docker`? I haven't heard of it before.

Reply:
https://pkgs.org/download/podman-docker
That's a wrapper script (provided by podman I believe) to have podman as a (for most stuff) drop-in replacement for docker

Reply:
Ah.. interesting. Definitely seems to be useful in cases where you can't use podman out of the box. I still don't agree however with us using this. Some people are running podman on their systems, so using `podman-docker` would be unnecessary overhead.

The existing makefile in opendatahub-operator master[1] already uses podman as the default. A couple other repositories are also using podman as default (following the same pattern of allowing users to specify docker if they want to)[2], [3]. Given that this would be a change in existing behavior I still think we should update (restore) usage of podman instead of docker.

Just for additional context, there is the upstream kserve/modelmesh project that uses docker by default[4], but allows developers to specify podman instead. 

[1] https://github.com/opendatahub-io/opendatahub-operator/blob/master/Makefile#L8
[2] https://github.com/opendatahub-io/notebooks/blob/main/Makefile#L1
[3] https://github.com/opendatahub-io/odh-manifests/blob/master/tests/Makefile (This actually doesn't even let you use docker without updating the makefile.
[4] https://github.com/kserve/modelmesh-serving/blob/main/Makefile#L20

Reply:
Well in that case we can't just replace docker with podman, right ?

(I know it does not work for everything, but for the limited stuff used in operator-sdk Makefiles I don't think there is anything that breaks ?)

Reply:
This conversation can be simplified to are we standardizing on `podman` or `docker` to build containers that run in OpenShift environments AND who has the burden of typing the extra characters for `make IMAGE_BUILDER=<podman|docker>`

Reply:
Yes, agree with moving to `podman` as default. For operator, I have update the IMAGE_BUILDER